assembly/src/parsers/ast/context.rs

```diff
@@ -310,6 +314,51 @@ impl ParserContext {
         Ok(())
     }
 
+    // DOCUMENTATION PARSERS
+    // ================================================================================================
+
+    pub fn parse_docs(&mut self, source: &str) -> Result<(), AssemblyError> {
+        let current_state = AsmSourceState::Empty;
+        let mut comments = String::new();
+
+        for line in source.lines() {
+            let new_state = parse_new_state(line);
+            if new_state != current_state {
+                match new_state {
+                    AsmSourceState::Func => {
+                        let func_name = get_proc_name(line);
+                        let value = self
+                            .local_procs
+                            .get_mut(&func_name)
+                            .expect("Undefined procedure");
+                        value.1.docs = if comments.is_empty() {
+                            None
+                        } else {
+                            Some(comments.clone())
+                        };
+                        comments.clear();
+                    }
+                    AsmSourceState::Comment => {
+                        comments.push_str(&remove_prefix(DOC_COMMENT_PREFIX, line));
+                        comments.push(' ')
```

Comment:
I think we should still retain line breaks within comments. So, instead of pushing a space at the end, we should probably push `\n` character.

Also, I'd probably wrap a comment into a struct which would look something like:
```Rust
pub struct Comment(String)
```

And then I would add a method to this struct to append a new line. Something like:
```Rust
pub fn append_line(&mut self, line: &str)
```
This method internally would strip out prefix, trim the line, and append it to the internal string.